src/corpora/auto.py

```diff
@@ -38,15 +38,23 @@ def get_auto_dataset(
     # Sanity check on input args
     stride = seq_len if stride < 0 else stride
     assert stride <= seq_len, f"Data grouping stride ({stride}) is smaller than sequence length: we are losing data."
-    if dataset_dir is not None:
-        file_names = os.listdir(dataset_dir)
-        file_type = os.path.splitext(file_names[0])[1][1:]
-        file_type = "json" if file_type == "jsonl" else file_type
-        dataset_files = {}
-        dataset_files["train"] = [f"{dataset_dir}/{fn}" for fn in file_names if "train" in fn]
-        dataset_files["validation"] = [f"{dataset_dir}/{fn}" for fn in file_names if "validation" in fn]
+
+    if local_dataset is not None:
+        # Fail fast if the train or validation splits do not have a single file
+        assert len(local_dataset["train_files"]) > 0, "Expected at least one file path for the train split"
+        assert len(local_dataset["validation_files"]) > 0, "Expected at least one file path for the validation split"
+
+        # Extracts dataset files locally
+        path_to_check: str = local_dataset["train_files"][0]
+        file_extension: str = get_file_extension(path_to_check)
+        file_extension = "json" if file_extension == "jsonl" else file_extension
+
+        dataset_files: Dict[str, List[str]] = {
+            "train": local_dataset["train_files"],
```

Comment:
might be worth supporting globs or something, though really I'd prefer moving away from this in favor of just writing HF datasets.